src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const resultArray = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateClone, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
```

Comment:
Move all cases to the object.

Example of usage: 
`case ActionTypes.RemoveProperties:`